tools/autotools/tsk_logical_imager.cpp

```diff
@@ -0,0 +1,749 @@
+/*
+ ** tsk_logical_imager
+ ** The Sleuth Kit 
+ **
+ ** Brian Carrier [carrier <at> sleuthkit [dot] org]
+ ** Copyright (c) 2010-2011 Brian Carrier.  All Rights reserved
+ **
+ ** This software is distributed under the Common Public License 1.0
+ **
+ */
+
+#include <iostream>
+#include <string>
+#include <algorithm>
+#include <locale>
+#include <codecvt>
+#include <direct.h>
+#include <winsock2.h>
+#include <locale.h>
+#include <Wbemidl.h>
+#pragma comment(lib, "wbemuuid.lib")
+
+#include <comutil.h>
+
+#include "tsk/tsk_tools_i.h"
+#include "tsk/auto/tsk_case_db.h"
+#include "tsk/img/img_writer.h"
+
+std::wstring GetLastErrorStdStrW();
+std::string GetErrorStdStr(DWORD err);
+std::wstring GetErrorStdStrW(DWORD err);
+
+static TSK_TCHAR *progname;
+
+static std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
+
+/**
+* toUpper: convert string to uppercase
+* @param srcStr to convert
+* @return uppercase string
+*/
+string toUpper(const string &srcStr)
+{
+	string outStr(srcStr);
+	std::transform(srcStr.begin(), srcStr.end(), outStr.begin(), ::toupper);
+
+	return outStr;
+}
+
+/**
+* toLower: convert string to lowercase
+* @param srcStr to convert
+* @return lowercase string
+*/
+string toLower(const string &srcStr)
+{
+	string outStr(srcStr);
+	std::transform(srcStr.begin(), srcStr.end(), outStr.begin(), ::tolower);
+
+	return outStr;
+}
+
+/**
+* Convert from UTF-16 to UTF-8.
+* Returns empty string on error
+*/
+string toNarrow(const std::wstring& a_utf16Str)
+{
+	try {
+		std::string narrow = converter.to_bytes(a_utf16Str);
+		return narrow;
+	}
+	catch (...) {
+		std::exception_ptr eptr = std::current_exception();
+		return "";
+	}
+}
+
+/**
+* Convert from UTF-8 to UTF-16.
+* Returns empty string on error
+*/
+std::wstring toWide(const string& a_utf8Str)
+{
+	try {
+		std::wstring wide = converter.from_bytes(a_utf8Str);
+		return wide;
+	}
+	catch (...) {
+		std::exception_ptr eptr = std::current_exception();
+		return L"";
+	}
+}
+
+/**
+* GetErrorStdStr - returns readable error message for the given error code
+*
+* @param err error code
+* @returns error message string
+*/
+string GetErrorStdStr(DWORD err) {
+	return toNarrow(GetErrorStdStrW(err));
+}
+
+/**
+* GetLastErrorStdStrW - returns readable widestring error message for the last error code as reported by GetLastError()
+*
+* @returns error message wide string
+*/
+std::wstring GetLastErrorStdStrW() {
+	DWORD error = GetLastError();
+	return GetErrorStdStrW(error);
+}
+
+/**
+* GetErrorStdStrW - returns readable widestring error message for the given error code
+*
+* @param err error code
+* @returns error message wide string
+*/
+std::wstring GetErrorStdStrW(DWORD a_err)
+{
+	if (ERROR_SUCCESS != a_err)
+	{
```

Comment:
There are inconsistencies in block format throughout the file. We generally put the open brace on the same line as the conditional.